### PR TITLE
Add image for DSK

### PR DESCRIPTION
**Pull request by SarahNadeau**
This PR adds DSK, a tool for k-mer counting in fasta or fastq files. It also adds test-dsk.yml and deploy-dsk.yml workflows. There are 2 commits that toggle from deploying DSK to my own docker hub to setting it up to deploy to StaPH-B's docker hub. So, here's where the image currently lives: https://hub.docker.com/repository/docker/snads/dsk. I'd love it to live in the StaPH-B docker hub moving forward, the deploy workflow is now set up for that.

- [x] Build your own docker image using a Dockerfile
  - [x] Directory structure should be name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. spades/3.12.0/Dockerfile)
  - [x] Includes the recommended LABELS
- [x] (Optional) Dockerfile is built with best practices and has been approved by a linter (such as https://hadolint.github.io/hadolint/)
- [x] Edit main README.md
- [x] Edit Program_Licenses.md
- [x] Create a simple container-specific README.md in the same directory as the Dockerfile (i.e. spades/3.12.0/README.md)
- [x] Write a GitHub actions workflow
  - [x] Should be located in .github/workflows/ and named test-<tool>.yml (i.e. .github/workflows/test-spades.yml)
  - [x] Any files required for building are located in the same directory as the Dockerfile (i.e. spades/3.12.0/my_spades_tests.sh)
  - [x] Have successfully run the workflow "Test <program name> image" in your forked repository
 See https://github.com/SarahNadeau/docker-builds/actions/runs/2028098727.

### CODE REVIEW

**Comment by SarahNadeau** [comment]
I am not sure why the test workflow was not run on pull request as specified [here](https://github.com/StaPH-B/docker-builds/pull/320/files#diff-a9628ef8e5ed24d5c1e5998605eaf864ba87a29b1118f1816ec7d434987060b7R6). Does the workflow show up as needing approval to a maintainer?

**Comment by erinyoung** [comment]
I looked into why it's not triggering a github action.

1. It hasn't attempted the github action
2. It isn't asking for approval to start a github action

File: .github/workflows/test-dsk.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - "dsk/0.0.100/Dockerfile"
+
+jobs:
+
+  build-to-test:
+    uses: StaPH-B/docker-builds/.github/workflows/build-to-test.yml@master
```
Comment (erinyoung) [inline .github/workflows/test-dsk.yml:13]: If you changed this to `./.github/workflows/build-to-test.yml`, does it trigger a workflow?

Comment (SarahNadeau) [reply .github/workflows/test-dsk.yml]: Indeed it does, I wonder why though. I'll take another read through the workflow docs. Anyways, should we update the template file accordingly here? https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/test-template.yml#L19

Also I did find in the docs that the workflow approval is only triggered the first time someone contributes, after that workflows run automatically.

**Comment by erinyoung** [comment]
That appears to have worked. I think we need to use the relative path.

**Comment by erinyoung** [comment]
In the meantime, could you add a small readme at `dsk/0.0.100/README.md`?

Generally when people get stuck on a container, they ask us how to use it instead of the container maintainer, and it makes our lives easier if there's a readme to refer them to.

**Review by erinyoung — APPROVED** [review]

**Pull request merged**